### PR TITLE
[wallet] improve local currency exchange rate polling

### PR DESCRIPTION
**Pull request by jeanregisser**
### Description

Made the local currency exchange rate polling not rely on `setTimeout` which triggers a warning on Android when the delay is greater than 1 minute. 

See https://github.com/facebook/react-native/blob/cc35d0f1bc0371e1e9d6b117daa9ee0862d00e74/Libraries/Core/Timers/JSTimers.js#L49-L53

### Tested

Added tests.

### Other changes

- `LOCAL_CURRENCY_SYMBOL` is now set during tests. Snapshots were updated accordingly.
- Fix "Confirming…" status color on exchange items.
- Auto refresh loop delay moved to the end of the loop. Prevents edge case of balance updating a little too late.

### Related issues

- Follow up for #870

### Backwards compatibility

Yes.

### CODE REVIEW

File: packages/mobile/src/localCurrency/saga.ts [hunk 1]
```diff
@@ -57,10 +58,17 @@ function* updateLocalCurrencyRate() {
 
   while (true) {
     try {
-      yield put(fetchCurrentRate())
+      const lastSuccessfulUpdate: number | undefined = yield select(
+        (state: RootState) => state.localCurrency.lastSuccessfulUpdate
+      )
+      if (lastSuccessfulUpdate && Date.now() - lastSuccessfulUpdate < MIN_UPDATE_INTERVAL) {
+        yield take()
```
Comment (jmrossy) [inline packages/mobile/src/localCurrency/saga.ts]: what does this do? Does this just take any action that's dispatched?

File: packages/mobile/src/localCurrency/saga.test.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  fetchCurrentRateFailure,
+  fetchCurrentRateStart,
+  fetchCurrentRateSuccess,
+} from 'src/localCurrency/actions'
+import { fetchExchangeRate, updateLocalCurrencyRate } from 'src/localCurrency/saga'
+import rootReducer from 'src/redux/reducers'
+import { createMockStore } from 'test/utils'
+
+const now = Date.now()
+Date.now = jest.fn(() => now)
```
Comment (jmrossy) [inline packages/mobile/src/localCurrency/saga.test.ts:12]: We should probably restore this after your tests are done

File: packages/mobile/src/localCurrency/saga.ts (hunk 1 above)
Comment (jeanregisser) [reply packages/mobile/src/localCurrency/saga.ts]: Yes.

File: packages/mobile/src/localCurrency/saga.test.ts (hunk 2 above)
Comment (jeanregisser) [reply packages/mobile/src/localCurrency/saga.test.ts:12]: Yes I wondered about this too. I borrowed this from another [test we already had](https://github.com/celo-org/celo-monorepo/blob/b7ad7840e6d37594a89dad2177b7eab6786150cc/packages/mobile/src/stableToken/saga.test.ts#L14).
From what I understand, tests within a file are isolated from other test files we have, so this is fine. What do you think?

File: packages/mobile/src/localCurrency/saga.ts (hunk 1 above)
Comment (jeanregisser) [reply packages/mobile/src/localCurrency/saga.ts]: What this does is check the date difference on every state change and only trigger the update after 24 hours.

Comment (jmrossy) [reply packages/mobile/src/localCurrency/saga.ts]: That seems wasteful, why check again of EVERY state change? 
Can we just use the approach we talked about the other day of only checking once per app session?

File: packages/mobile/src/localCurrency/saga.test.ts (hunk 2 above)
Comment (jmrossy) [reply packages/mobile/src/localCurrency/saga.test.ts:12]: Ok np then

File: packages/mobile/src/localCurrency/saga.ts (hunk 1 above)
Comment (jeanregisser) [reply packages/mobile/src/localCurrency/saga.ts]: It might feel wasteful, but this is not different from what happens with `connect` on the component side. Each `mapStateToProps` is called on every state change.

I wanted this to be resilient to long running sessions (app in background and MainActivity remaining active).
I think a good alternative approach would be to use [AppState](https://facebook.github.io/react-native/docs/appstate) and trigger this time difference check when the app state becomes `active`.
What do you think?

Comment (jmrossy) [reply packages/mobile/src/localCurrency/saga.ts]: This still feels like overkill to support a 1% case. I don't think I prefer the AppState approach either, as I don't feel the complexity is worth the benefit.
I _would_ be open to having this run when a user triggers a refresh (e.g. pulling down on wallet home).
https://github.com/celo-org/celo-monorepo/blob/master/packages/mobile/src/home/saga.ts#L36

I believe the auto-refresh will have this run relatively often which should meet the requirement you're going for here

Comment (jeanregisser) [reply packages/mobile/src/localCurrency/saga.ts]: It's not a 1% case. I believe it will happen to most users. 

Not a big fan of now coupling this with the balance auto refresh either. 😸 
It uses a 10 secs timer loop. This could be seen as wasteful too when the app is not in use.
We might as well do the same in that module if we go that way.

Comment (jmrossy) [reply packages/mobile/src/localCurrency/saga.ts]: once every 10 seconds seems better than every state change right? 
I'm still imagining you'd keep the `lastSuccessfulUpdate` check.
It seems like we're going around a bit on this, let's chat about it tmrw morning :)

**Review by jmrossy — DISMISSED** [review]

File: packages/mobile/src/localCurrency/saga.ts [hunk 3]
```diff
@@ -48,27 +42,24 @@ async function fetchExchangeRate(symbol: string): Promise<number> {
...
   }
 }
 
+export function* watchFetchCurrentRate() {
+  yield takeLeading(Actions.FETCH_CURRENT_RATE, fetchLocalCurrencyRateSaga)
+}
+
 export function* localCurrencySaga() {
-  yield spawn(updateLocalCurrencyRate)
+  yield spawn(watchFetchCurrentRate)
```
Comment (jmrossy) [inline packages/mobile/src/localCurrency/saga.ts:64]: There's no reason to spawn a separate watcher here, may as well inline the `takeLeading`

File: packages/mobile/src/home/saga.ts [hunk 4]
```diff
@@ -83,3 +88,4 @@ export function* homeSaga() {
 }
 
 export const _watchRefreshBalances = watchRefreshBalances
+export const _autoRefreshSaga = autoRefreshSaga
```
Comment (jmrossy) [inline packages/mobile/src/home/saga.ts:91]: Related to our discussion on slack, I stopped using this pattern for now b.c. of the import order issue. Volpe (who originally suggested this) agrees.

**Review by jmrossy — APPROVED** [review]
Two small comments but otherwise LGTM

File: packages/mobile/src/localCurrency/saga.ts (hunk 3 above)
Comment (jeanregisser) [reply packages/mobile/src/localCurrency/saga.ts:64]: Main reason is giving clarity into what it does and testing that specific logic.

File: packages/mobile/src/home/saga.ts (hunk 4 above)
Comment (jeanregisser) [reply packages/mobile/src/home/saga.ts:91]: Noted 👍 Leaving this one for now given it was already used in this file and I don't want make more unrelated changes.
